### ISSUE TITLE
PHP Fatal error in your plugin

### ISSUE THREAD

**Issue by Beee4life** [subharanjanm/debug-bar-actions-and-filters-addon#24]
I got this error from your plugin.

FYI: I am not using Jetpack.

```
PHP Fatal error:  Uncaught Error: Cannot use object of type Automattic\Jetpack\Autoloader\jpbbce79c8bf878b60e2b54dfa0e05ac16\Shutdown_Handler as array in /var/www/html/web/app/plugins/debug-bar-actions-and-filters-addon/debug-bar-action-and-filters-addon.php:185
```

**Comment by jrfnl** [comment]
This plugin has no relation to JetPack either. The error however indicated that JetPack would be running on your install and that JetPack is doing something funky with a hook-in.

**Comment by Beee4life** [comment]
hence why I thought I'd report it, since I'm not touching jetpack with a 10 ft pole :)

**Comment by jrfnl** [comment]
Well, unless there is some way to create a scenario with which I could reproduce the issue, there's not much which can be done.

**Comment by subharanjanm** [comment]
@Beee4life Could you please help us reproduce the issue at our end ?

**Comment by Beee4life** [comment]
By the looks of it, I suspect it comes from a Woocommerce check, since Woocommerce advises to use Jetpack.

We do use woocommerce in the site where the error occurred but no jetpack.

I don't have a sandbox installation yet on this macbook so have yet to test it in a clean environment, but I think the result will be the same...

So I think it can be recreated by installing woocommerce. Would have to double check though.

**Comment by mqudsi** [comment]
After upgrading JetPack, I now see this error (again for `Shutdown_Handler`) as well. I don't have WooCommerce installed.

I'm not sure what this plugin does is perfectly safe, though. Here's line 185:

```php
} elseif ( ( is_array( $single_function['function'] ) || is_object( $single_function['function'] ) ) && dbafa_is_closure( $single_function['function'][0] ) ) {
```

If it's an object, there's no guarantee that `$single_function['function'][0]` is safe to be indexed/dereferenced like that. 

I'm not entirely confident that this is the best workaround, but this at least doesn't crash and still reports the registered callback:

```diff
--- /tmp/.psub.A3Cljn5uzE       2021-12-15 13:41:22.096701000 -0600
+++ /var/www/neosmart.net/wordpress/wp-content/plugins/debug-bar-actions-and-filters-addon/debug-bar-action-and-filters-addon.php       2021-12-15 13:40:11.474108000 -0600
@@ -182,7 +182,7 @@
                                                // Type 2 - closure
                                                $table .= '<li>[<em>' . esc_html__( 'closure', 'debug-bar-actions-and-filters-addon' ) . '</em>]</li>';
                                                $signature = get_class( $single_function['function'] ) . $hook_in_count;
-                                       } elseif ( ( is_array( $single_function['function'] ) || is_object( $single_function['function'] ) ) && dbafa_is_closure( $single_function['function'][0] ) ) {
+                                       } elseif ( ( is_array( $single_function['function'] ) ) && dbafa_is_closure( $single_function['function'][0] ) ) {
                                                // Type 3 - closure within an array
                                                $table .= '<li>[<em>' . esc_html__( 'closure', 'debug-bar-actions-and-filters-addon' ) . '</em>]</li>';
                                                $signature = get_class( $single_function['function'] ) . $hook_in_count;
@@ -205,6 +205,7 @@
                                        } else {
                                                // Type 8 - undetermined
                                                $table .= '<li><pre>' . var_export( $single_function, true ) . '</pre></li>';
+                                               $signature = var_export($single_function, true);
                                        }

                                        $hook_in_count ++;
@@ -233,4 +234,4 @@
        </div>';

        return $output;
-}
\ No newline at end of file
+}
```

**Comment by subharanjanm** [comment]
This has been fixed in version 1.5.5.
